app/controllers/entry_controller.rb

```diff
@@ -4,13 +4,10 @@ class EntryController < ApplicationController
   before_filter :load_entry, :only => [:comment]
 
   def comment
-    if @challenge.participator?(current_user) && params[:comment] && params[:comment][:text].size > 0
+    if @challenge.participator?(current_user) && @entry && params[:comment].try(:[], :text).try(:present?)
```

Comment:
thoughts on using `fetch` with a default? may make things a little more clear. up to you